project/driver_controls/inc/driver_io.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+// Provides an interface to configure the GPIO pins for the driver control devices.
+// This allows a cleaner device initialization than having to initialize each pin with
+// different settings
+
+#include "gpio.h"
+#include "interrupt.h"
+
+typedef void (*DriverIOCallback)(GPIOAddress *address, void *context);
+
+typedef enum {
+  DRIVER_IO_POWER_SWITCH = 0,
+  DRIVER_IO_GAS_PEDAL,
+  DRIVER_IO_DIRECTION_SELECTOR,
+  DRIVER_IO_CRUISE_CONTROL,
+  DRIVER_IO_CRUISE_CONTROL_INC,
+  DRIVER_IO_CRUISE_CONTROL_DEC,
+  DRIVER_IO_TURN_SIGNAL,
+  DRIVER_IO_HAZARD_LIGHT,
+  DRIVER_IO_MECHANICAL_BRAKE,
+  NUM_DRIVER_IO_INPUTS
+} DriverIO;
+
+typedef struct DriverIODevice {
```

Comment:
If you aren't allowing them to pass in structs, there's no point in exposing it in your header.